http-server/src/test/java/io/airlift/http/server/TestHttpServerProvider.java

```diff
@@ -359,6 +359,55 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    @Test
+    public void testClientCertificatePem()
+            throws Exception
+    {
+        HttpServlet servlet = new HttpServlet()
+        {
+            @Override
+            protected void doGet(HttpServletRequest request, HttpServletResponse response)
+                    throws IOException
+            {
+                X509Certificate[] certs = (X509Certificate[]) request.getAttribute("javax.servlet.request.X509Certificate");
+                if ((certs == null) || (certs.length == 0)) {
+                    throw new RuntimeException("No client certificate");
+                }
+                if (certs.length > 1) {
+                    throw new RuntimeException("Received multiple client certificates");
+                }
+                X509Certificate cert = certs[0];
+                response.getWriter().write(cert.getSubjectX500Principal().getName());
+                response.setStatus(HttpServletResponse.SC_OK);
+            }
+        };
+
+        config.setHttpEnabled(false)
+                .setAdminEnabled(false)
+                .setHttpsEnabled(true)
+                .setHttpsPort(0)
+                .setKeystorePath(getResource("clientcert-pem/server.pem").getPath())
+                .setKeystorePassword("airlift")
+                .setTrustStorePath(getResource("clientcert-pem/ca.crt").getPath());
+
+        createAndStartServer(servlet);
+
+        HttpClientConfig clientConfig = new HttpClientConfig()
+                .setKeyStorePath(getResource("clientcert-pem/client.pem").getPath())
+                .setKeyStorePassword("airlift")
+                .setTrustStorePath(getResource("clientcert-pem/ca.crt").getPath());
+
+        System.out.println(httpServerInfo.getHttpsUri());
+        try (JettyHttpClient httpClient = new JettyHttpClient(clientConfig)) {
```

Comment:
Maybe add a helper method `assertClientCertificateRequest` as this code is the same as above